freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/RebootInstancesService.java

```diff
@@ -0,0 +1,111 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.logger.MDCBuilder;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.instance.InstanceStatus;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.health.NodeHealthDetails;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.reboot.RebootInstancesRequest;
+import com.sequenceiq.freeipa.controller.exception.BadRequestException;
+import com.sequenceiq.freeipa.entity.InstanceMetaData;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.flow.instance.InstanceEvent;
+import com.sequenceiq.freeipa.service.freeipa.flow.FreeIpaFlowManager;
+import com.sequenceiq.freeipa.service.stack.instance.InstanceMetaDataService;
+
+@Service
+public class RebootInstancesService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RebootInstancesService.class);
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private FreeIpaFlowManager flowManager;
+
+    @Inject
+    private FreeIpaHealthDetailsService healthDetailsService;
+
+    @Inject
+    private InstanceMetaDataService instanceMetaDataService;
+
+    private Map<String, InstanceStatus> getInstanceHealth(String accountId, RebootInstancesRequest request) {
+        if (!request.isForceReboot()) {
+            return healthDetailsService.getHealthDetails(request.getEnvironmentCrn(), accountId).getNodeHealthDetails().stream()
+                    .collect(Collectors.toMap(NodeHealthDetails::getInstanceId, NodeHealthDetails::getStatus));
+        }
+        return Collections.emptyMap();
+    }
+
+    private Set<String> getInvalidInstances(Collection<String> allInstances, RebootInstancesRequest request) {
+        Consumer<String> logit = id -> {
+            if (!allInstances.contains(id)) {
+                LOGGER.info("Unknown instance ID {} in request", id);
+            }
+        };
+        return Stream.ofNullable(request.getInstanceIds()).flatMap(Collection::stream)
+                .peek(logit)
+                .filter(id -> !allInstances.contains(id))
+                .collect(Collectors.toSet());
+    }
+
+    private Map<String, InstanceMetaData> getInstancesToReboot(Map<String, InstanceMetaData> allInstances,
+            List<String> instanceIds, Map<String, InstanceStatus> health) {
+        Consumer<Map.Entry> logit = entry -> {
+            if (health.get(entry.getKey()) != null && health.get(entry.getKey()).isAvailable()) {
+                LOGGER.info("Not rebooting AVAILABLE instance {} because force reboot was not selected.", entry.getKey());
+            }
+        };
+        return allInstances.entrySet().stream()
+                .peek(logit)
+                .filter(entry -> instanceIds == null || instanceIds.isEmpty() || instanceIds.contains(entry.getKey()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
I think something is wrong here. You used the heath to filter what to log, but you don't use it to filter. In my opinion peek should just log and don't have any logic at all, this could be misleading

Reply:
refactored to for loops instead of streams because of the split nature of logging the items to be filtered out.